application-ideas-ui/src/main/resources/Ideas/IdeasJSON.xml

```diff
@@ -0,0 +1,66 @@
+<?xml version="1.1" encoding="UTF-8"?>
+
+<!--
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+-->
+
+<xwikidoc version="1.3" reference="Ideas.IdeasJSON" locale="">
+  <web>Ideas</web>
+  <name>IdeasJSON</name>
+  <language/>
+  <defaultLanguage/>
+  <translation>0</translation>
+  <creator>xwiki:XWiki.Admin</creator>
+  <parent>Ideas.WebHome</parent>
+  <author>xwiki:XWiki.Admin</author>
+  <contentAuthor>xwiki:XWiki.Admin</contentAuthor>
+  <version>1.1</version>
+  <title>IdeasJSON</title>
+  <comment/>
+  <minorEdit>false</minorEdit>
+  <syntaxId>xwiki/2.1</syntaxId>
+  <hidden>false</hidden>
+  <content>{{include reference="XWiki.LiveTableResultsMacros" /}}
+
+{{velocity wiki="false"}}
+#if("$!{request.xpage}" == 'plain')
+  $response.setContentType('application/json')
+#end
+##==================
+## Initialize the map that will store the JSON data.
+##==================
+#set($map = {})
+##==================
+## Build the JSON in memory using the previous map
+##==================
+#gridresult_buildJSON("$!request.classname" $request.collist.split(',') $map)
+##==================
+## Modify the JSON data from the map or add new data.
+##==================
+#foreach($row in $map.get('rows'))
+  #set($rowDocRef = $services.model.resolveDocument($row.get('doc_fullName')))
+  ## Add an additional constraint for the edit right (just an example).
+  #set($discard = $row.put('doc_rest_url', $services.ideas.url($rowDocRef)))
+#end
```

Comment:
Honestly I think this is too much overhead just to get the REST URL. If you need the REST URL only in JavaScript then you should simply drop the ``getRestURL`` method from the ``IdeasManager`` and script service and write it in JavaScript. You have everything needed there. This means dropping this custom live table results page also, and the additional live table column.